### ISSUE TITLE
Flickering on first zoom out after clearing map data layer

### ISSUE THREAD

**Issue by westnordost** [tangrams/tangram-es#2133]
**Summary**
Zooming in and out usually looks smooth. However, if you clear a map data layer while being zoomed in, the first time you zoom out again will lead to flickering of the features in the map data layer that should have been cleared.

It looks as if the features in the map data layer have not been cleared yet on the different zoom levels and are so only when first entering that zoom level.
I did not test whether this bug only occurs when clearing a map data layer or also when setting (different) features on it.

Video here: https://www.westnordost.de/misc/tangram-flicker-2132.mp4

**Reproduction**
- add a map data layer via `MapController.addDataLayer` and add a geometry to it
- Zoom in
- Clear the map data layer
- Zoom out. Doesn't matter if manually via gesture or with an animation.

**Result**
The cleared features in the map data layer flicker on zooming out.

**Expected result**
They do not flicker but are gone right away.

**Environment**
Tangram Version 0.11.2
Tested on Android Version 5.1.2
Tested on Sony Xperia Z1 Compact

**Comment by matteblair** [comment]
Just a clarification: in https://github.com/tangrams/tangram-es/issues/2131 it sounded like this started in version 0.12.0, is that the case?

**Comment by westnordost** [comment]
No, #2131 existed all-along, at least since 0.9.6 but was fixed in 0.12.0.

This issue turned up on 0.12.0.

**Comment by tallytalwar** [comment]
Seems like a consequence of the changes as part of #2118, possible proxy tiles are holding onto this data. If I understand this xorrextly, @matteblair I might steel assignee from you.

**Comment by RubenKelevra** [comment]
I just want to throw in an example how it looks like, running on Android 9:

https://youtu.be/NYgtkI5tU7U
